### PR TITLE
Reality Check Index

### LINKED ISSUE #4: FE: Search Results Reality Check
As a visitor, when I'm on the search results page
I should see a "Reality Check" button at the bottom
When I click it, a number field appears and I'm prompted to enter my monthly income
When I enter my monthly income, and click the "submit" button, I should see a new set of results that are only listings I could hypothetically afford

### PR DESCRIPTION
**Pull request by zach-bergman**
From search results page, reality check button at bottom of page opens a form to enter monthly income. When submit button is clicked, directs to reality check index page with filtered results.

reality_check method in service and facade to handle the api request.
RealityCheckController with index action to pass filtered Listing objects to index view.
Index view styled the same as the search index view.

Tests for everything, some sad path tests added into the service spec, not sure if needed. Let me know what you think and if anything needs to be changed!

### CODE REVIEW

File: app/views/search/index.html.erb [hunk 1]
```diff
@@ -48,5 +48,21 @@
...
   </div>
-  <button class="btn btn-danger btn-lg reality-check-btn">Reality Check</button>
+
+<!-- Button -->
+<div class="rc-button">
+  <button type="button" class="btn btn-danger btn-lg reality-check-btn" data-toggle="collapse" data-target="#incomeFormContainer" aria-expanded="false" aria-controls="incomeFormContainer" data-turbolinks="false">Reality Check</button>
+</div>
+
+<!-- Collapsible Form -->
+<div class="collapse" id="incomeFormContainer" style="margin-top: 20px;">
```
Comment (RodrigoACG) [inline app/views/search/index.html.erb:58]: Amazing job on all this Styling!!!

**Review by RodrigoACG — APPROVED** [review]

File: app/controllers/reality_check_controller.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class RealityCheckController < ApplicationController
+  def index
+    @facade = OomprBeFacade.new
+    cookies[:monthly] = params[:monthly]
+    @rc_listings = @facade.reality_check(cookies[:city], cookies[:monthly])
+  end
```
Comment (jdmchugh111) [inline app/controllers/reality_check_controller.rb:6]: thanks for making this cookie as we had discussed during inception!

File: app/views/reality_check/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,56 @@
+<div class="container container-custom">
+  <h1 class="oompr-title"><%= link_to 'oompr', root_path, class: 'oompr-title-link' %></h1>
+  <div class="row">
+    <div class="col-md-6 offset-md-3">
+      <div class="listings-container">
+        <% if @rc_listings.empty? %>
+          <div class="alert alert-warning" role="alert">
+            No listings found. Want to search for rentals in your area?
+          </div>
```
Comment (jdmchugh111) [inline app/views/reality_check/index.html.erb:9]: thanks for including this rental prompt - not sure if we'll have the chance to implement it as part of our mvp, but thank you for being so detail oriented

File: app/controllers/reality_check_controller.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class RealityCheckController < ApplicationController
+  def index
```
Comment (jdmchugh111) [inline app/controllers/reality_check_controller.rb:2]: also, interesting design choice to have a separate reality check controller, but I like it. Puts less conditional logic into the controller

File: app/views/search/index.html.erb [hunk 5]
```diff
@@ -48,5 +48,21 @@
...
+  <div class="card card-body">
+    <%= form_with url: reality_check_index_path, method: :get, local: true do |form| %>
+      <div class="form-group">
+        <%= form.label :monthly, "Enter your monthly income:" %>
+        <%= form.number_field :monthly, class: "form-control", required: true, min: 0 %>
+      </div>
+      <%= form.submit "Submit", class: "btn btn-primary" %>
+    <% end %>
```
Comment (jdmchugh111) [inline app/views/search/index.html.erb:66]: great use of form with url to redirect and also pass a param here

**Review by jdmchugh111 — APPROVED** [review]
awesome work again!

**Pull request merged**